gen/builders/tipsets.go

```diff
@@ -0,0 +1,79 @@
+package builders
+
+import (
+	"github.com/filecoin-project/test-vectors/schema"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+
+	"github.com/filecoin-project/go-address"
+)
+
+// TipsetSeq is a sequence of tipsets to be applied during the test.
+// TipsetSeq provides methods to build a sequence of tipsets, such as
+// adding new tipsets and null rounds.
+type TipsetSeq struct {
+	tipsets  []*Tipset
+	epoch    abi.ChainEpoch
+	messages []*ApplicableMessage
+}
+
+type Tipset struct {
+	tss *TipsetSeq
```

Comment:
why do you need the back reference?

Reply:
To add the block messages to the messages index, which is later used by the asserter.